test/tools/metroconfig.test.ts

```diff
@@ -108,9 +112,116 @@ describe('metroconfig', () => {
+    });
+
+    test('removes Web Replay when platform is web and includeWebReplay is false', () => {
+      const modifiedConfig = excludeSentryWebReplay(config, false);
+      const result = resolveRequest(modifiedConfig, contextMock, '@sentry/replay', 'web');
+
+      expect(result).toEqual({ type: 'empty' });
+      expect(originalResolverMock).not.toHaveBeenCalled();
+    });
+
+    /* Test expected to fail but not required since excludeSentryWebReplay is not run when includeWebReplay is true.
+    test('keep Web Replay when platform is android and includeWebReplay is true', () => {
+      const modifiedConfig = excludeSentryWebReplay(config, true);
+      resolveRequest(modifiedConfig, contextMock, '@sentry/replay', 'android');
+
+      expect(originalResolverMock).toHaveBeenCalledWith(contextMock, '@sentry/replay', 'android');
+    });
+*/
+
+    test('removes Web Replay when platform is android and includeWebReplay is false', () => {
+      const modifiedConfig = excludeSentryWebReplay(config, false);
+      const result = resolveRequest(modifiedConfig, contextMock, '@sentry/replay', 'android');
+
+      expect(result).toEqual({ type: 'empty' });
+      expect(originalResolverMock).not.toHaveBeenCalled();
+    });
+
+    test('removes Web Replay when platform is android and includeWebReplay is undefined', () => {
+      const modifiedConfig = excludeSentryWebReplay(config, undefined);
+      const result = resolveRequest(modifiedConfig, contextMock, '@sentry/replay', 'android');
+
+      expect(result).toEqual({ type: 'empty' });
+      expect(originalResolverMock).not.toHaveBeenCalled();
+    });
+
+    /* Test expected to fail but not required since excludeSentryWebReplay is not run when includeWebReplay is true.
+    test('keep Web Replay when platform is ios and includeWebReplay is true', () => {
+      const modifiedConfig = excludeSentryWebReplay(config, true);
+      resolveRequest(modifiedConfig, contextMock, '@sentry/replay', 'ios');
+
+      expect(originalResolverMock).toHaveBeenCalledWith(contextMock, '@sentry/replay', 'ios');
+    });
+*/
+
+    test('removes Web Replay when platform is ios and includeWebReplay is false', () => {
+      const modifiedConfig = excludeSentryWebReplay(config, false);
+      const result = resolveRequest(modifiedConfig, contextMock, '@sentry/replay', 'ios');
+
+      expect(result).toEqual({ type: 'empty' });
+      expect(originalResolverMock).not.toHaveBeenCalled();
+    });
+
+    test('removes Web Replay when platform is ios and includeWebReplay is undefined', () => {
+      const modifiedConfig = excludeSentryWebReplay(config, undefined);
+      const result = resolveRequest(modifiedConfig, contextMock, '@sentry/replay', 'ios');
+
+      expect(result).toEqual({ type: 'empty' });
+      expect(originalResolverMock).not.toHaveBeenCalled();
+    });
+
+    test('calls originalResolver when moduleName is not @sentry/replay', () => {
```

Comment:
Can you add tests for indludeReplay=false and module not @sentry/replay?

At the moment an empty module would be returned always.